packages/ui/Touchable/tests/Touchable-specs.js

```diff
@@ -175,6 +250,76 @@ describe('Touchable', () => {
 
 			expect(actual).toBe(expected);
 		});
+
+		test('should call onDragStart event', (done) => {
+			const Component = Touchable(DivComponent);
+			const handler = jest.fn();
+			const {rerender} = render(<Component onDrag={() => {}} />);
+
+			const component = screen.getByTestId('component');
+
+			fireEvent.mouseDown(component, {clientX: 10, clientY: 20});
+			act(() => jest.advanceTimersByTime(20));
+
+			rerender(<Component onDrag={() => {}} onDragStart={handler} />);
+
+			jest.runOnlyPendingTimers();
+
+			fireEvent.mouseMove(component, {clientX: 20, clientY: 30});
+			act(() => jest.advanceTimersByTime(20));
+			fireEvent.mouseMove(component, {clientX: 30, clientY: 40});
+			act(() => jest.advanceTimersByTime(20));
+			fireEvent.mouseMove(component, {clientX: 40, clientY: 50});
+			fireEvent.mouseUp(component, {clientX: 40, clientY: 50});
+
+			expect(handler).toHaveBeenCalled();
+			done();
+		});
+
+		test('should call onDragEnd event', (done) => {
+			const Component = Touchable(DivComponent);
+			const handler = jest.fn();
+			const {rerender} = render(<Component onDrag={() => {}} />);
+
+			const component = screen.getByTestId('component');
+
+			fireEvent.mouseDown(component, {clientX: 10, clientY: 20});
+			act(() => jest.advanceTimersByTime(20));
+			fireEvent.mouseMove(component, {clientX: 20, clientY: 30});
+			act(() => jest.advanceTimersByTime(20));
+			fireEvent.mouseMove(component, {clientX: 30, clientY: 40});
+			act(() => jest.advanceTimersByTime(20));
+			fireEvent.mouseMove(component, {clientX: 40, clientY: 50});
+			rerender(<Component onDrag={() => {}} onDragEnd={handler} />);
+
+			jest.runOnlyPendingTimers();
+
+			fireEvent.mouseUp(component, {clientX: 40, clientY: 50});
+			expect(handler).toHaveBeenCalled();
+			done();
+		});
```

Comment:
Please add the following test to cover mouseLeave on drag

```suggestion
				test('should not call onDragEnd on mouseLeave', (done) => {
				const Component = Touchable(DivComponent);
				const handler = jest.fn();
				const {rerender} = render(<Component onDrag={() => {}} />);

				const component = screen.getByTestId('component');

				fireEvent.mouseDown(component, {clientX: 10, clientY: 20});
				act(() => jest.advanceTimersByTime(20));
				fireEvent.mouseMove(component, {clientX: 20, clientY: 30});
				act(() => jest.advanceTimersByTime(20));
				fireEvent.mouseMove(component, {clientX: 30, clientY: 40});
				act(() => jest.advanceTimersByTime(20));
				fireEvent.mouseMove(component, {clientX: 40, clientY: 50});
				rerender(<Component onDrag={() => {}} onDragEnd={handler} />);

				jest.runOnlyPendingTimers();

				fireEvent.mouseLeave(component);
				expect(handler).not.toHaveBeenCalled();
				done();
		});
```